compiler/rustc_pattern_analysis/src/rustc.rs

```diff
@@ -463,6 +463,11 @@ impl<'p, 'tcx: 'p> RustcMatchCheckCtxt<'p, 'tcx> {
                     _ => bug!("pattern has unexpected type: pat: {:?}, ty: {:?}", pat, ty),
                 };
             }
+            PatKind::DerefPattern { .. } => {
+                // FIXME(deref_patterns): At least detect that `box _` is irrefutable.
+                fields = vec![];
```

Comment:
You've since added `arity` that needs to be bound

Reply:
oh yeah good point